backend/ventserver/protocols/server.py

```diff
@@ -137,9 +146,30 @@ def output(self) -> Optional[ReceiveOutputEvent]:
         if not any_updated:
             return None
 
+        # Kill frontend process if it stops responding.
+        # The frontend service will automatically restart the frontend process.
+        if (int(self.current_time - self.last_frontend_event) > 1) and\
+            (int(self.current_time - self.frontend_connection_time) > 2) and\
+            self.frontend_connected:
+            if self._kill_process is None or\
```

Comment:
Need suggestions on better way to do this.

Reply:
Would it be enough to only check for `int(self.current_time - self.last_frontend_event) > 1)`? So the server protocol only cares about whether it's been too long since it received a websocket message, and the integration code and/or I/O endpoint for running a shell command only cares about running the command when the server tells it to (and the integration code also cares about ignoring any timeout signals which occur too soon after the last timeout signal).

Reply:
No it's not enough. In the situations where the system has just started or the frontend has restarted, the `last_frontend_event` has it's default value or the old value. So, as soon as the frontend connects, the `current_time` is compared with, not so updated, `last_frontend_event`. And since the difference is usually more than a sec, the kill process is spawned. Furthermore, the `last_front_event` is then never updated; the system is stuck in a situation where whenever the frontend connects the backend kills it.